src/engine/BMGame.php

```diff
@@ -3266,27 +3081,26 @@ protected function get_playerDataArray($requestingPlayerIdx) {
         $playerDataArray = array();
 
         // helper arrays of data that BMGame naturally produces in array form
-        $roundScoreArray = $this->get__roundScoreArray();
         $sideScoreArray = $this->get_sideScoreArray();
         $canStillWinArray = $this->get_canStillWinArray();
 
-        foreach ($this->getBMPlayerProps('playerId') as $playerIdx => $playerId) {
+        foreach ($this->playerArray as $playerIdx => $player) {
             $playerData = array(
-                'playerId'            => $playerId,
+                'playerId'            => $player->playerId,
                 'button'              => $this->get_buttonInfo($playerIdx),
                 'activeDieArray'      => $this->get_activeDieArray($playerIdx, $requestingPlayerIdx),
                 'capturedDieArray'    => $this->get_capturedDieArray($playerIdx),
                 'outOfPlayDieArray'   => $this->get_outOfPlayDieArray($playerIdx),
                 'swingRequestArray'   => $this->get_swingRequestArray($playerIdx, $requestingPlayerIdx),
-                'optRequestArray'     => $this->get_optRequestArray($playerIdx),
-                'prevSwingValueArray' => $this->get_prevSwingValueArray($playerIdx),
-                'prevOptValueArray'   => $this->get_prevOptValueArray($playerIdx),
+                'optRequestArray'     => $player->optRequestArray,
+                'prevSwingValueArray' => $player->prevSwingValueArray,
+                'prevOptValueArray'   => $player->prevOptValueArray,
                 'waitingOnAction'     => $this->get_waitingOnActionArray($playerIdx, $requestingPlayerIdx),
```

Comment:
Yeah, hmm, right, because you don't want BMPlayer to be in charge of the lying.  In principle you might break those into a BMPlayer accessor which does the accessing, and a BMGame routine which takes the clean value given by BMPlayer and filters it based on requesting player and game state.  But i don't feel strongly.

You should add a code comment here explaining this distinction, though.